go-controller/pkg/ovn/policy.go

```diff
@@ -300,16 +301,28 @@ func (oc *DefaultNetworkController) syncNetworkPolicies(networkPolicies []interf
 	}
 
 	stalePGs := []string{}
-	err := oc.addressSetFactory.ProcessEachAddressSet(func(addrSetName, namespaceName, policyName string) error {
-		if policyName != "" && !expectedPolicies[namespaceName][policyName] {
-			// policy doesn't exist on k8s. Delete the port group
-			portGroupName := fmt.Sprintf("%s_%s", namespaceName, policyName)
-			hashedLocalPortGroup := hashedPortGroup(portGroupName)
-			stalePGs = append(stalePGs, hashedLocalPortGroup)
-			// delete the address sets for this old policy from OVN
-			if err := oc.addressSetFactory.DestroyAddressSetInBackingStore(addrSetName); err != nil {
-				klog.Errorf(err.Error())
-				return err
+	err := oc.addressSetFactory.ProcessEachAddressSet(func(hashedName, addrSetName string) error {
+		// netpol name has format fmt.Sprintf("%s.%s.%s.%d", gp.policyNamespace, gp.policyName, direction, gp.idx)
+		s := strings.Split(addrSetName, ".")
+		sLen := len(s)
+		// priority should be a number
+		_, numErr := strconv.Atoi(s[sLen-1])
+		if sLen >= 4 && (s[sLen-2] == "ingress" || s[sLen-2] == "egress") && numErr == nil {
+			// address set is owned by network policy
+			// namespace doesn't have dots
+			namespaceName := s[0]
+			// policyName may have dots, join in that case
+			policyName := strings.Join(s[1:sLen-2], ".")
+			if !expectedPolicies[namespaceName][policyName] {
```

Comment:
is this dangerous? Could there be cases where the `namespaceName` is not present in expectedPolicies map and then we try to extract `policyName` from nil?

Reply:
in other languages yeah, but in golang this does not cause a crash if the key does not exist, just returns a zero value :)